### ISSUE TITLE
Modular Modding

### ISSUE THREAD

**Issue by RealTelefonmast** [Profugo-Barbatus/TiberiumRim#28]
Some people agree with the idea of making our mod modular, so basically having a core mod and adding all additions like factions in a seperate mod.
I already had this idea before but now I also think that it could be rather hard to do -
as we build everything ontop of each other.

So having agressive Tiberium would ruin gameplay for those not having the factions mod (and the additional Tiberium removing/preventing buildings). -

I'm not sure about the whole idea and how we could make it work so everyone can be happy.

So basically, shall we make this modular? 
Yes or no?

**Comment by Profugo-Barbatus** [comment]
The plan was already a certain level of modularity (You can have Tiberium without the faction pack, but you cannot have the faction pack without Tiberium, for example).

Hugslib provides the functionality for a mod config panel. This will allow us to compartmentalize function, while still offering granular control. Players could be given config settings that, depending on how much power Hugslib gives me, will allow fine tuned control of the game.

This of course, would require locking our project to Hugslib for the time being. In of itself, that shouldn't be a problem, as Hugslib is lightweight enough to likely stay updated, and offers some other convenient tools.

If you agree to this course of action, I'll lay out a 1.2 version plan, which will involve Hugslib integration, the code adjustments needed, and configuration options for players.

**Comment by RealTelefonmast** [comment]
Sounds like a plan - as much as I want our mod to be independent though, but you say that hugslib is likely to stay updated so I suppose it should not be a problem - either way, this could be pretty handy afterall.

So I will agree with this.
